### DISCUSSION TITLE
Users Questions from Wechat

### DISCUSSION THREAD

**Discussion by silviahsia1201** [Oneflow-Inc/oneflow#9058] [Q&A] [answered]
咱们（One Flow）的验证图和训练图需要分别定义为独立的两个graph的原因是什么？

**Comment by strint** [comment, marked as answer]
静态图的采用了全图的 trace 机制，trace 就是根据代码的执行算子的路径去构图。

因为训练和验证的执行逻辑不同，所以就需要 trace 成两个不同的图。

这两个图也适合做不同的配置。比如训练图可以配置 optimizer、grad scale，而预测图则不需要。

两个图会通过共同的nn.Module来共享参数。
